### PR TITLE
cost-management - API changes

### PR DESCRIPTION
**Pull request by yaborens**
…anagement/preview/2018-08-01-preview/costmanagement.json

Add more Scopes to Alerts.
Also, add /updateStatus to Alerts

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by KrisBash — APPROVED** [review]
Changes seem fine. one question in comments

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 1]
```diff
@@ -1860,13 +1860,486 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
     "/subscriptions/{subscriptionId}/providers/Microsoft.CostManagement/alerts/{alertId}": {
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: we do have API to get all subscription alerts:
"/subscriptions/{subscriptionId}/providers/Microsoft.CostManagement/alerts"

Get RG alerts:
"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CostManagement/alerts"

**Comment by yaborens** [comment]
Who can merge the code? as I'm not authorized to merge this pull request

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 2]
```diff
@@ -1860,13 +1860,486 @@
...
+          },
+		  {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          },
+          {
+            "$ref": "#/parameters/departmentIdParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter alerts by properties/definition/type, properties/definition/category, properties/definition/criteria, properties/costEntityId, properties/creationTime, properties/closeTime, properties/status, properties/source. Supported operators are 'eq','lt', 'gt', 'le', 'ge'.",
```
Comment (praries880) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Examples corresponding to this change?

Also, is the list of filters a closed list? If so... why not use an enum?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 3]
```diff
@@ -1860,13 +1860,486 @@
...
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter alerts by properties/definition/type, properties/definition/category, properties/definition/criteria, properties/costEntityId, properties/creationTime, properties/closeTime, properties/status, properties/source. Supported operators are 'eq','lt', 'gt', 'le', 'ge'.",
```
Comment (praries880) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: @yaborens
Examples corresponding to this change?

Also, is the list of filters a closed list? If so... why not use an enum?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 4]
```diff
@@ -1860,13 +1860,486 @@
...
+          },
+		  {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          },
+          {
+            "$ref": "#/parameters/enrollmentAccountIdParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter alerts by properties/definition/type, properties/definition/category, properties/definition/criteria, properties/costEntityId, properties/creationTime, properties/closeTime, properties/status, properties/source. Supported operators are 'eq','lt', 'gt', 'le', 'ge'.",
```
Comment (praries880) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: @yaborens
Examples corresponding to this change?

Also, is the list of filters a closed list? If so... why not use an enum?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 5]
```diff
@@ -1860,13 +1860,486 @@
...
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/managementGroupIdParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter alerts by properties/definition/type, properties/definition/category, properties/definition/criteria, properties/costEntityId, properties/creationTime, properties/closeTime, properties/status, properties/source. Supported operators are 'eq','lt', 'gt', 'le', 'ge'.",
```
Comment (praries880) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: @yaborens
Examples corresponding to this change?

Also, is the list of filters a closed list? If so... why not use an enum?

**Review by praries880 — CHANGES_REQUESTED** [review]
@yaborensth Some comments

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 2 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Examples corresponding to this change? yes
is the list of filters a closed list? If so... why not use an enum? this is how all other groups implemented a filter.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 3 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: same comments as above

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 4 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: same comments as above

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 5 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: same comments as above

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 3 above)
Comment (praries880) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: I see only this change in the example linked here : https://github.com/Azure/azure-rest-api-specs/pull/4818/files/1d5c0565121e5fef4897b32e78318df919406717#diff-cd2cdd86afbc1836331eaef83f213b9cR30

Is that all changes in the example for this new API?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 2 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Hi,
Is that all that changes there for the new API added here? yes

- Filter is not a close group, thus it cannot be enum.
- properties/definition/type - not a close group, thus it cannot be enum.
- properties/definition/category - Already an Enum
- properties/definition/criteria - Already an Enum
- properties/costEntityId - not a close group, thus it cannot be enum.
- properties/creationTime- Not an Enum field
- properties/closeTime - Not an Enum field
- properties/status - Already an Enum
- properties/source - Already an Enum

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 3 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: yes

**Review by praries880 — APPROVED** [review]

**Pull request merged**